server/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java

```diff
@@ -159,51 +125,66 @@ private void removeNodesFromCluster(String[] args)
-              + "and after the process of remove-datanode is over, "
+              + "and after the process of removing datanode is over, "
               + "you are supposed to delete directory and data of the removed-datanode manually");
     }
   }
 
   /**
-   * fetch all datanode info from ConfigNode, then compare with input 'ips'
+   * fetch all datanode info from ConfigNode, then compare with input 'args'
    *
-   * @param endPorts data node ip:port, split with ','
+   * @param args datanode id or ip:rpc_port
    * @return TDataNodeLocation list
    */
-  private List<TDataNodeLocation> buildDataNodeLocations(String endPorts)
-      throws BadNodeUrlException {
+  private List<TDataNodeLocation> buildDataNodeLocations(String args) {
     List<TDataNodeLocation> dataNodeLocations = new ArrayList<>();
-    if (endPorts == null || endPorts.trim().isEmpty()) {
+    if (args == null || args.trim().isEmpty()) {
       return dataNodeLocations;
     }
 
-    List<TEndPoint> endPoints = NodeUrlUtils.parseTEndPointUrls(endPorts);
-
-    try (ConfigNodeClient client = new ConfigNodeClient()) {
-      dataNodeLocations =
-          client.getDataNodeConfiguration(-1).getDataNodeConfigurationMap().values().stream()
-              .map(TDataNodeConfiguration::getLocation)
-              .filter(location -> endPoints.contains(location.getClientRpcEndPoint()))
-              .collect(Collectors.toList());
-    } catch (TException e) {
-      logger.error("Get data node locations failed", e);
+    // Now support only single datanode deletion
+    if (args.split(",").length > 1) {
+      logger.info("Incorrect input format, usage: <id>/<ip>:<rpc-port>");
+      return dataNodeLocations;
     }
 
-    if (endPoints.size() != dataNodeLocations.size()) {
-      logger.error(
-          "Build DataNode locations error, "
-              + "because number of input DataNode({}) not equals the number of fetched DataNodeLocations({}), "
-              + "please check your IP:RPC_PORT inputs",
-          endPoints.size(),
-          dataNodeLocations.size());
-      dataNodeLocations.clear();
-      return dataNodeLocations;
+    // Below supports multiple datanode deletion, split by ',', and is reserved for extension
+    try {
+      List<TEndPoint> endPoints = NodeUrlUtils.parseTEndPointUrls(args);
+      try (ConfigNodeClient client = new ConfigNodeClient()) {
+        dataNodeLocations =
+            client.getDataNodeConfiguration(-1).getDataNodeConfigurationMap().values().stream()
+                .map(TDataNodeConfiguration::getLocation)
+                .filter(location -> endPoints.contains(location.getClientRpcEndPoint()))
+                .collect(Collectors.toList());
+      } catch (TException e) {
+        logger.error("get data node locations failed", e);
+      }
+    } catch (BadNodeUrlException e) {
+      try (ConfigNodeClient client = new ConfigNodeClient()) {
+        for (String id : args.split(",")) {
```

Comment:
Now we only support removing one datanode id, if it's necessary to using `args.split(",")`

Reply:
Reserving multi-node deletion logic is suggested by boss Qiao~